File: preview/hosted_numbers/authorization_document/create-default/create-default.5.x.rb
# Download the helper library from https://www.twilio.com/docs/ruby/install
require 'rubygems'
require 'twilio-ruby'

# Your Account Sid and Auth Token from twilio.com/console
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'
@client = Twilio::REST::Client.new(account_sid, auth_token)

authorization_document = @client.preview
  .hosted_numbers
  .authorization_documents
  .create(
     hosted_number_order_sids: ['hosted_number_order_sids'],
     address_sid: 'ADXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX',
     email: 'email',
     contact_title: 'contact_title',
     contact_phone_number: 'contact_phone_number'
   )

puts authorization_document.sid

File: preview/hosted_numbers/authorization_document/create-default/create-default.5.x.cs
// Install the C# / .NET helper library from twilio.com/docs/csharp/install

using System;
using Twilio;
using Twilio.Converters;
using Twilio.Rest.Preview.HostedNumbers;


class Program 
{
    static void Main(string[] args)
    {
        // Find your Account Sid and Token at twilio.com/console
        // DANGER! This is insecure. See http://twil.io/secure
        const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
        const string authToken = "your_auth_token";

        TwilioClient.Init(accountSid, authToken);

        var authorizationDocument = AuthorizationDocumentResource.Create(
            hostedNumberOrderSids: Promoter.ListOfOne("hosted_number_order_sids"),
            addressSid: "ADXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
            email: "email",
            contactTitle: "contact_title",
            contactPhoneNumber: "contact_phone_number"
        );

        Console.WriteLine(authorizationDocument.Sid);
    }
}
